src/components/Requirements/CompletedSubReqCourse.vue

```diff
@@ -37,94 +37,55 @@
-      this.setCourseAndSemesterForNonTransferCredits(crseTaken);
-    } else {
-      this.setCourseAndSemesterForTransferCredits(crseTaken);
-    }
+    courseTaken: { type: Object as PropType<CourseTaken>, required: true },
   },
   computed: {
     semesters(): readonly FirestoreSemester[] {
       return store.state.semesters;
     },
-    courseLabel() {
+    courseLabel(): string {
       return `Course ${this.subReqCourseId + 1}`;
     },
-    resetText() {
+    resetText(): string {
       return 'Reset';
     },
-    semesterLabel() {
-      let label = `in ${this.$data.semesterType} ${this.$data.semesterYear}`;
+    isTransferCredit(): boolean {
+      return this.courseTaken.subject === 'AP' || this.courseTaken.subject === 'IB';
+    },
+    courseColorAndSemesterLabel(): { readonly semesterLabel: string; readonly color: string } {
       if (this.isTransferCredit) {
-        label = 'in Transfer Credits';
+        return { semesterLabel: 'Transfer Credits', color: transferCreditColor };
       }
-      return label;
-    },
-  },
-  methods: {
-    dragListener(event: { preventDefault: () => void }) {
-      if (!this.$data.scrollable) event.preventDefault();
-    },
-    setCourseAndSemesterForNonTransferCredits(crseTaken: CourseTaken) {
-      const courseTakenCode = `${crseTaken.subject} ${crseTaken.number}`;
+      const courseTakenCode = `${this.courseTaken.subject} ${this.courseTaken.number}`;
       for (let i = 0; i < this.semesters.length; i += 1) {
         const semester = this.semesters[i];
         const filteredSemesterCourses = semester.courses.filter(
-          course => course.crseId === crseTaken.courseId && course.code === courseTakenCode
+          course => course.crseId === this.courseTaken.courseId && course.code === courseTakenCode
         );
         if (filteredSemesterCourses.length > 0) {
-          const course = filteredSemesterCourses[0];
-          const [subject, number] = course.code.split(' ');
-          this.color = course.color;
-          this.courseSubject = subject;
-          this.courseNumber = number;
-          this.courseUniqueId = course.uniqueID;
-          this.semesterType = semester.type;
-          this.semesterYear = semester.year;
-          break;
+          return {
+            semesterLabel: `${semester.type} ${semester.year}`,
+            color: filteredSemesterCourses[0].color,
+          };
         }
       }
+      return { semesterLabel: `${getCurrentSeason()} ${getCurrentYear()}`, color: '' };
     },
-    setCourseAndSemesterForTransferCredits(crseTaken: CourseTaken) {
-      this.color = this.transferCreditColor;
-      this.courseSubject = crseTaken.subject;
-      this.courseNumber = crseTaken.number;
-    },
+  },
+  methods: {
     onReset() {
-      this.$emit('deleteCourseFromSemesters', this.courseUniqueId);
+      this.$emit(
+        'deleteCourseFromSemesters',
+        this.isTransferCredit ? 0 : this.courseTaken.courseId
+      );
```

Comment:
Resetting doesn't seem to work. I think it's because in Dashboard.vue, the `deleteCourseFromSemesters()` method still uses `uniqueID`. 


https://user-images.githubusercontent.com/54298311/108152331-d7a92280-70a6-11eb-8fc3-d7af95550488.mov

Reply:
should be fixed now